### PR TITLE
build(ingest): upgrade to sqlalchemy 1.4, drop 1.3 support

### PR DESCRIPTION
**Pull request by mayurinehate**
## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py [hunk 1]
```diff
@@ -451,17 +451,10 @@ def _get_operation_aspect_work_unit(
                 yield wu
 
     def _process_snowflake_history_row(
-        self, row: Any
+        self, event_dict: dict
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py:454]: We use DictCursor in snowflake, so this is always a dictionary.

File: docs/how/updating-datahub.md [hunk 2]
```diff
@@ -5,6 +5,7 @@ This file documents any backwards-incompatible changes in DataHub and assists pe
 ## Next
 
 ### Breaking Changes
+- #8810 - Removed support for SQLAlchemy 1.3.x . Only SQLAlchemy 1.4.x is supported now.
```
Comment (hsheth2) [inline docs/how/updating-datahub.md:8]:
```suggestion
- #8810 - Removed support for SQLAlchemy 1.3.x. Only SQLAlchemy 1.4.x is supported now.
```

File: metadata-ingestion/setup.py [hunk 3]
```diff
@@ -112,7 +112,7 @@ def get_long_description():
 
 sql_common = {
     # Required for all SQL sources.
-    "sqlalchemy>=1.3.24, <2",
+    "sqlalchemy>=1.4.0, <2",
```
Comment (hsheth2) [inline metadata-ingestion/setup.py:115]: Do we actually support 1.4.0? We probably want a stronger lower bound

```suggestion
    "sqlalchemy>=1.4.0, <2",
```

File: metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_usage_v2.py:454]: nice

**Review by hsheth2 — APPROVED** [review]

File: metadata-ingestion/setup.py (hunk 3 above)
Comment (mayurinehate) [reply metadata-ingestion/setup.py]: I noticed lint failure in ./tests/test_helpers/sqlalchemy_mypy_plugin.py when using sqlalchemy==1.4.0. Upgrading to 1.4.8 lower bound solves this. Do you have any thoughts on how to decide stronger lower bound  - maybe based on recency OR quickly try out some sqlalchemy versions in test matrix and choosing the least which works .

Comment (hsheth2) [reply metadata-ingestion/setup.py]: I suspect our true lower bound is higher than 1.4.8, but I don't think it's worth the effort of finding it just yet

For now, let's set the lower bound to something that's a year or so old (maybe 1.4.39 from Jun 2022) and add a comment explaining that it's a temporary lower bound that we're open to loosening/tightening as requirements show up

main compat issues I anticipate are gonna be from GE, so let's see how it goes

Comment (mayurinehate) [reply metadata-ingestion/setup.py]: updated lower bound to 1.4.39 and added comment in this [commit](https://github.com/datahub-project/datahub/pull/8810/commits/70a00b39e84418e3c0917f3c7819b7944bd0a6b8).

**Review by hsheth2 — APPROVED** [review]

**Pull request merged**